src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -5227,6 +5108,35 @@ static void Main()
         }
 
         [Fact]
+        public void VerifyDiscardIL()
+        {
+            var source =
+@"
+class C
+{
+    static int M()
+    {
+        var (x, _, _) = (1, new C(), 2);
+        return x;
+    }
+}
+";
+
+            var comp = CompileAndVerify(source, additionalRefs: s_valueTupleRefs);
+            comp.VerifyDiagnostics();
+            comp.VerifyIL("C.M()", @"
+{
+  // Code size        8 (0x8)
+  .maxstack  1
+  .locals init (C V_0)
+  IL_0000:  newobj     ""C..ctor()""
+  IL_0005:  stloc.0
```

Comment:
I'm not sure why the result of `new C()` is stored into a local here. I'll chat with Vlad.